imports/client/components/PuzzleListPage.tsx

```diff
@@ -424,37 +428,52 @@ interface PuzzleListPageProps extends PuzzleListPageWithRouterParams {
   canUpdate: boolean;
   allPuzzles: PuzzleType[];
   allTags: TagType[];
+  hunt?: HuntType;
 }
 
 class PuzzleListPage extends React.Component<PuzzleListPageProps> {
   render() {
+    const leadLinks = this.props.hunt?.homepageUrl && (
+      <p className="lead">
+        <a href={this.props.hunt.homepageUrl} target="_blank" rel="noopener noreferrer" title="Open the hunt homepage">
+          <FontAwesomeIcon icon={faExternalLinkAlt} />
+          {' '}
+          Hunt homepage
+        </a>
+      </p>
+    );
     if (!this.props.ready) {
       return <span>loading...</span>;
     } else {
       return (
-        <PuzzleListView
-          match={this.props.match}
-          history={this.props.history}
-          location={this.props.location}
-          huntId={this.props.match.params.huntId}
-          canAdd={this.props.canAdd}
-          canUpdate={this.props.canUpdate}
-          puzzles={this.props.allPuzzles}
-          allTags={this.props.allTags}
-        />
+        <div>
+          {leadLinks}
+          <PuzzleListView
+            match={this.props.match}
+            history={this.props.history}
+            location={this.props.location}
+            huntId={this.props.match.params.huntId}
+            canAdd={this.props.canAdd}
+            canUpdate={this.props.canUpdate}
+            puzzles={this.props.allPuzzles}
+            allTags={this.props.allTags}
+          />
+        </div>
       );
     }
   }
 }
 
 const PuzzleListPageContainer = withTracker(({ match }: PuzzleListPageWithRouterParams) => {
+  const huntHandle = Meteor.subscribe('mongo.hunts', { _id: match.params.huntId });
   const puzzlesHandle = Meteor.subscribe('mongo.puzzles', { hunt: match.params.huntId });
   const tagsHandle = Meteor.subscribe('mongo.tags', { hunt: match.params.huntId });
 
   // Don't bother including this in ready - it's ok if it trickles in
   Meteor.subscribe('subscribers.counts', { hunt: match.params.huntId });
 
-  const ready = puzzlesHandle.ready() && tagsHandle.ready();
+  const ready = huntHandle.ready() && puzzlesHandle.ready() && tagsHandle.ready();
+  const hunt = Hunts.findOne({ _id: match.params.huntId });
```

Comment:
Go ahead and make this:

```
// Hunt is guaranteed to have been loaded by HuntApp already.
const hunt = Hunts.findOne({ _id: match.params.huntId })!;
```

The `!` at the end asserts to typescript that this will not be `undefined`.